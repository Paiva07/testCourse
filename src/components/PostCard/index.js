const PostCard = ({ title, cover, body, id }) => {
  return (
    <div>
      <img src={cover} alt={title} />
      <h2>
        {title} {id}
      </h2>
      <p>{body}</p>
    </div>
  );
};
export default PostCard;
