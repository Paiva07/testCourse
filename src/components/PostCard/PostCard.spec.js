import { render, screen } from '@testing-library/react';
import PostCard from '.';

const props = {
  title: 'title',
  body: 'body',
  id: 1,
  cover: 'img/img.png',
};

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);
    expect(screen.getByRole('img', { name: props.title })).toBeInTheDocument(); //Validamento se Existe no doc
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.cover,
    ); //Validando se tem o atributo src com o cover
    expect(
      screen.getByRole('heading', { name: 'title 1' }),
    ).toBeInTheDocument();
    expect(screen.getByText(props.body)).toBeInTheDocument(); //Valida se tem o texto em algum local
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />); //Pegando o primeiro elemento do comp.
    expect(container.firstChild).toMatchSnapshot(); //Valida o componente com o SnapShot
  });
});
