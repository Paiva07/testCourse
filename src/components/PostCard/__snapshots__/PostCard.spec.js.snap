// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div>
  <img
    alt="title"
    src="img/img.png"
  />
  <h2>
    title
     
    1
  </h2>
  <p>
    body
  </p>
</div>
`;
